### PR TITLE
Add R CMD check workflow

### PR DESCRIPTION
**Pull request by eitsupi**
Based on discussion https://github.com/pola-rs/r-polars/issues/14#issuecomment-1379472243

### CODE REVIEW

**Comment by eitsupi** [comment]
Sorry, I meant to test it on my fork first, but I made a mistake when creating this PR.......

**Comment by sorhawell** [comment]
wow thanks  :) I will take a look at it

**Comment by sorhawell** [comment]
> Sorry, I meant to test it on my fork first, but I made a mistake when creating this PR.......
I dont mind, it is not like there are that many jobs running yet

**Comment by eitsupi** [comment]
> wow thanks :) I will take a look at it

Thanks, I will be back here when I have success with my fork, so please wait a while.

**Comment by eitsupi** [comment]
`CI=false` works fine for R-release.
But this requires installation script of the apt dependency installation for the R-devel on ubuntu workflow.
Do you want tests for R-devel?
If it is not needed at this time, R-devel can be commented to keep the workflow simple.

**Comment by eitsupi** [comment]
The workflow now works. So, I think we can simplify the pkgdown workflow. (I can do this in this PR if you like.)

**Comment by sorhawell** [comment]
Many thanks. My pkgdown.yaml workflow is not very pretty and I love to get some help setting it up better. I don't mind using pak if it can work.

**Comment by sorhawell** [comment]
the many deprecation warnings come from this action: https://github.com/actions-rs/toolchain/issues/216
it seems the cool rust people switch to [dtolnay/rust-toolchain@stable](https://github.com/actions-rs/toolchain/issues/216)

This could also be addressed separately if out of scope of your PR

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            ${{ runner.os }}-cargo-target
+
+      - run: echo "rust-toolchain=nightly" >>"$GITHUB_ENV"
+      - run: echo "rust-toolchain=nightly-gnu" >>$env:GITHUB_ENV
+        if: runner.os == 'Windows'
+
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust-toolchain }}
```
Comment (eitsupi) [inline .github/workflows/check.yaml:62]: > the many deprecation warnings come from this action: https://github.com/actions-rs/toolchain/issues/216
> it seems the cool rust people switch to https://github.com/actions-rs/toolchain/issues/216

Yes, of course I use it here 👌.

**Pull request merged**